tests/D2L.CodeStyle.Analyzers.Test/Specs/StatelessFuncAnalyzerSpec.cs

```diff
@@ -85,6 +107,41 @@ internal sealed class Usages {
 			};
 
 			var func = new StatelessFunc<int>( /* StatelessFuncIsnt( Invocations are not allowed: evil() ) */ evil() /**/ );
+			AttributeFuncReceiver.Accept<int>( /* StatelessFuncIsnt( Invocations are not allowed: evil() ) */ evil() /**/ );
+		}
+
+		public void StatelessFunc() {
+			var func = new StatelessFunc<int>( new StatelessFunc<int>( () => 0 ) );
+			AttributeFuncReceiver.Accept( new StatelessFunc<int>( () => 0 ) );
+		}
+
+		public void StatelessFuncFromVar() {
+			var f = new StatelessFunc<int>( () => 0 );
+
+			var func = new StatelessFunc<int>( f );
+			AttributeFuncReceiver.Accept( f );
+		}
+
+		public void StatelessFuncFromParam( StatelessFunc<int> f) {
+			var func = new StatelessFunc<int>( f );
+			AttributeFuncReceiver.Accept( f );
+		}
+
+		public void StatelessFuncAttrParam( [StatelessFunc] Func<int> f ) {

```

Comment:
I guess this is a little broken since I could do `f = () => this.things`...?

Reply:
Maybe? But isn't this also one of the primary ways you wanted to be able to do this?

Reply:
Why the frig doesn't C# have readonly vars/parameters yet